### PR TITLE
mvvm first implementation

### PR DESCRIPTION
**Pull request by atrelinski**
This is data separation to its own structure, rather then mvvm implementation.
I would like to go with those changes since:
- it is tested (it behaves correctly, even there were few changes in workflow)
- it is enough to stop blocking next planned implementations
- too many changes at once will be hard to verify
To have mvvm, GraphControl should observe DataModel rather than directly use it.
ViewModel should observe DataModel and maybe even become part of it.

Please check proposed changes, if they are more or less OK,
I will remove commented out code and try to solve "DO NOT SUBMIT" items

### CODE REVIEW

**Comment by KrzysztofBuchacz** [comment]
You are right, my mistake, just ignore my comment about xaml.cs files.

File: DataModel.cs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    class DataModel
+    {
+        #region Members And Properties
+
+        /// <summary>
+        /// Holds point in time when entire build started (Solution).
+        /// </summary>
+        public DateTime StartTime { get; private set; }
+        public Dictionary<string, DateTime> CurrentBuilds { get; private set; } = new Dictionary<string, DateTime>(); //TODO: Check if collection can be changed even if private
+        public List<BuildInfo> FinishedBuilds { get; private set; } = new List<BuildInfo>(); //TODO: Check if collection can be changed even if private
```
Comment (KrzysztofBuchacz) [inline DataModel.cs]: Using collections as read only properties is more complicated, you should probably use ReadOnlyCollectionTemplate, example:
```suggestion
         private List <BuildInfo> _FinishedBuilds = new List<BuildInfo>();
         ReadOnlyCollection <BuildInfo> FinishedBuilds { get { _FinishedBuilds.AsReadOnly(); } } 
```

Comment (atrelinski) [reply DataModel.cs]: Should I rework that all members in all classes should begin from underscore (_)?
Right now all properties are Camel case, but members format is inconsistent (lowercase or Camel case)

File: DataModel.cs [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+        #endregion HelperMethods
+
+        #region CPU+HDDPerformance
+
+        void timer_Tick(object sender, ElapsedEventArgs e)
+        {
+            CollectPerformanceData(false);
+        }
+
+        //TODO: This makes CPU Average is wrong
```
Comment (KrzysztofBuchacz) [inline DataModel.cs]: Average is less precise but not wrong

**Pull request merged**